File: apps/web/public/.well-known/agon-agent/scripts/create_wallet.py
from __future__ import annotations

import argparse
import json

from lib.state import load_or_create_wallet, wallet_path


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("--state-dir", default="./.agon-agent")
    parser.add_argument("--role", choices=["primary", "sparring"], default="primary")
    args = parser.parse_args()

    wallet = load_or_create_wallet(args.state_dir, args.role)
    print(
        json.dumps(
            {
                "role": args.role,
                "wallet_path": str(wallet_path(args.state_dir, args.role)),
                "wallet_address": wallet["address"],
                "created": True,
            },
            indent=2,
        )
    )


if __name__ == "__main__":
    main()

File: apps/web/public/.well-known/agon-agent/scripts/create_wallet.js
#!/usr/bin/env node

const { parseArgs } = require('node:util');
const { createWallet } = require('./lib/wallet');
const { jsonResult, walletPath } = require('./lib/state');

function main() {
  const { values } = parseArgs({
    options: {
      'state-dir': { type: 'string', default: './.agon-agent' },
      role: { type: 'string', default: 'primary' },
      force: { type: 'boolean', default: false },
    },
  });

  const result = createWallet(values['state-dir'], values.role, values.force);
  process.stdout.write(`${JSON.stringify(jsonResult({
    state: 'wallet_ready',
    artifacts: {
      walletPath: walletPath(values['state-dir'], values.role),
    },
    data: {
      role: values.role,
      walletAddress: result.record.address,
      created: result.created,
      reused: result.reused,
    },
  }), null, 2)}\n`);
}

try {
  main();
} catch (error) {
  process.stderr.write(`${error.message}\n`);
  process.exit(1);
}
